src/mbgl/text/placement.cpp

```diff
@@ -518,6 +561,52 @@ void Placement::placeBucket(
         for (auto it = sortedSymbols.rbegin(); it != sortedSymbols.rend(); ++it) {
             placeSymbol(*it);
         }
+    } else if (mapMode == MapMode::Tile && !avoidEdges) {
+        // In this case we first try to place symbols, which intersects the tile borders, so that
+        // those symbols will remain even if each tile is handled independently.
+        const auto& symbolInstances = bucket.symbolInstances;
+        std::vector<std::reference_wrapper<const SymbolInstance>> sorted(symbolInstances.begin(),
+                                                                         symbolInstances.end());
+        optional<style::TextVariableAnchorType> variableAnchor;
+        if (!variableTextAnchors.empty()) variableAnchor = variableTextAnchors.front();
+        std::unordered_map<uint32_t, bool> sortedCache;
+        sortedCache.reserve(sorted.size());
+
+        auto intersectsTileBorder = [&](const SymbolInstance& symbol) -> bool {
+            if (symbol.textCollisionFeature.alongLine || symbol.textCollisionFeature.boxes.empty()) return false;
+
+            auto it = sortedCache.find(symbol.crossTileID);
+            if (it != sortedCache.end()) return it->second;
+
+            Point<float> offset{};
+            if (variableAnchor) {
+                const CollisionBox& textBox = symbol.textCollisionFeature.boxes.front();
+                float width = textBox.x2 - textBox.x1;
+                float height = textBox.y2 - textBox.y1;
+                offset = calculateVariableLayoutOffset(*variableAnchor,
+                                                       width,
+                                                       height,
+                                                       symbol.variableTextOffset,
+                                                       symbol.textBoxScale,
+                                                       rotateWithMap,
+                                                       pitchWithMap,
+                                                       state.getBearing());
+            }
+            bool result = collisionIndex.featureIntersectsTileBorders(
+                symbol.textCollisionFeature, offset, posMatrix, pixelRatio, *tileBorders);
+            sortedCache.insert(std::make_pair(symbol.crossTileID, result));
+            return result;
+        };
+
+        std::stable_sort(
+            sorted.begin(), sorted.end(), [&intersectsTileBorder](const SymbolInstance& a, const SymbolInstance& b) {
+                return intersectsTileBorder(a) && !intersectsTileBorder(b);
```

Comment:
Does this satisfy [`Compare`](https://en.cppreference.com/w/cpp/named_req/Compare ) requirements?